src/components/tl/mlx5/mcast/tl_mlx5_mcast_one_sided_reliability.c

```diff
@@ -0,0 +1,213 @@
+                                                          "one_sided_slots_buffer");
+    if (!comm->one_sided_slots_buffer) {
+        tl_error(comm->lib, "unable to malloc for one_sided_slots_buffer");
+        status = UCC_ERR_NO_MEMORY;
+        goto failed;
+    }
+    comm->one_sided_slots_mr = ibv_reg_mr(comm->ctx->pd, comm->one_sided_slots_buffer,
+                                          one_sided_total_slots_size, IBV_ACCESS_LOCAL_WRITE |
+                                          IBV_ACCESS_REMOTE_READ | IBV_ACCESS_REMOTE_WRITE);
+    if (!comm->one_sided_slots_mr) {
+        tl_error(comm->lib, "unable to register for one_sided_slots_mr");
+        status = UCC_ERR_NO_RESOURCE;
+        goto failed;
+    }
+    
+    /* this array holds local information about the slot status that was read from remote ranks */
+    comm->remote_slot_info = ucc_calloc(1, comm->commsize * ONE_SIDED_SLOTS_INFO_SIZE,
+                                        "remote_slot_info");
+    if (!comm->remote_slot_info) {
+        tl_error(comm->lib, "unable to malloc for remote_slot_info");
+        status = UCC_ERR_NO_MEMORY;
+        goto failed;
+    }
+    comm->remote_slot_info_mr = ibv_reg_mr(comm->ctx->pd, comm->remote_slot_info,
+                                           comm->commsize * ONE_SIDED_SLOTS_INFO_SIZE,
+                                           IBV_ACCESS_REMOTE_WRITE | IBV_ACCESS_LOCAL_WRITE |
+                                           IBV_ACCESS_REMOTE_READ);
+    if (!comm->remote_slot_info_mr) {
+        tl_error(comm->lib, "unable to register for remote_slot_info_mr");
+        status = UCC_ERR_NO_RESOURCE;
+        goto failed;
+    }
+
+    comm->one_sided_reliability_info = ucc_calloc(1, sizeof(ucc_tl_mlx5_one_sided_reliable_team_info_t) *
+                                                  comm->commsize, "one_sided_reliability_info");
+    if (!comm->one_sided_reliability_info) {
+        tl_error(comm->lib, "unable to allocate mem for one_sided_reliability_info");
+        status = UCC_ERR_NO_MEMORY;
+        goto failed;
+    }
+
+    status = ucc_tl_mlx5_mcast_create_rc_qps(comm->ctx, comm);
+    if (UCC_OK != status) {
+        tl_error(comm->lib, "RC qp create failed");
+        goto failed;
+    }
+
+    /* below holds the remote addr/rkey to local slot field of all the
+     * processes used in async protocol */
+    comm->one_sided_reliability_info[comm->rank].slot_mem.rkey        = comm->one_sided_slots_mr->rkey;
+    comm->one_sided_reliability_info[comm->rank].slot_mem.remote_addr = (uint64_t)comm->one_sided_slots_buffer;
+    comm->one_sided_reliability_info[comm->rank].port_lid             = comm->ctx->port_lid;
+    for (i = 0; i < comm->commsize; i++) {
+        comm->one_sided_reliability_info[comm->rank].rc_qp_num[i] = comm->mcast.rc_qp[i]->qp_num;
+    }
+
+    tl_debug(comm->lib, "created the allgather reliability structures");
+
+    return UCC_OK;
+
+failed:
+    if (comm->one_sided_slots_mr) {
```

Comment:
are we sure all those fields are set to NULL before entering the current function?

Reply:
yes, comm object is set to 0 after malloc.